### DISCUSSION TITLE
Adding to piggy Bank automatically - Rule

### DISCUSSION THREAD

**Discussion by bartekd123** [firefly-iii/firefly-iii#6480] [Firefly III]
I am trying to have a rule that adds to a piggy bank when a condition is met, but it isn't working. When I do my imports, I have rules that update a transaction "Convert the transaction to a transfer " . I also have another rule right after that "Add/remove transaction amount in piggy".

Is there a way to make this work?

**Comment by JC5** [comment]
Could be a bug, I'll check it out!

**Reply by bartekd123** [reply]
Did you want me to submit a bug for this one, or this discussion thread is enough for it to be on your list?

**Reply by JC5** [reply]
Just checked. I can't replicate it. Converting to a transfer and then add to a piggy works.

**Reply by bartekd123** [reply]
Sorry James, you are right. It wasn't working for me because I had a strict rule on the piggy bank depending on a category that was not yet applied I believe. I just tested what you said and it looks like it is working as expected. Thanks again!
